Adds Storage Starter Sample with Spring Resource

meltsufin opened the pull request.

joaoandremartins commented on spring-cloud-gcp-examples/spring-cloud-gcp-storage-resource-example/README.adoc:
```diff
@@ -0,0 +1,45 @@
+
+This code sample demonstrates how to read and write files with the Spring Resource abstraction for
+Google Cloud Storage using the
+link:../../spring-cloud-gcp-starters/spring-cloud-gcp-starter-storage/README.adoc[Spring Cloud GCP Storage Starter].
```
You can omit the /README.adoc bit.

joaoandremartins commented on spring-cloud-gcp-examples/spring-cloud-gcp-storage-resource-example/README.adoc:
```diff
@@ -0,0 +1,45 @@
+
+= Running the example
+
+1. Make sure that you have the Cloud SDK installed by following https://cloud.google.com/sdk/docs/[these instructions].
```
installed and initialized?

joaoandremartins commented on spring-cloud-gcp-examples/spring-cloud-gcp-storage-resource-example/README.adoc:
```diff
@@ -0,0 +1,45 @@
+
+1. Make sure that you have the Cloud SDK installed by following https://cloud.google.com/sdk/docs/[these instructions].
+
+2. Create a new globally unique bucket in Google Cloud Storage. You can use the `gsutil` command that comes with the
```
Just "Create a new bucket in Google Cloud Storage"?

joaoandremartins commented on spring-cloud-gcp-examples/spring-cloud-gcp-storage-resource-example/README.adoc:
````diff
@@ -0,0 +1,45 @@
+$ gsutil copy my-file.txt gs://$BUCKET
+```
+
+4. Edit the `src/main/java/com/example/WebController.java` so that `REPLACE_WITH_YOUR_BUCKET` is replaced
````
Edit the ... file?

joaoandremartins commented on spring-cloud-gcp-examples/spring-cloud-gcp-storage-resource-example/README.adoc:
```diff
@@ -0,0 +1,45 @@
+
+7. Navigate to http://localhost:8080/ in your browser to print the contents of the file you uploaded in 3.
+
+8. Now, try to do a POST to the same endpoint using `curl`:
```
To update the file contents, send a POST request to the same endpoint using `curl`

joaoandremartins commented on spring-cloud-gcp-examples/spring-cloud-gcp-storage-resource-example/pom.xml:
```diff
@@ -0,0 +1,25 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
```
Maybe unrelated to this PR, but I'm actually wondering whether our examples should have this parent.

It would probably be better not to, just so we could show some BOM usage, too.

meltsufin replied:
Valid point, but let's handle it separately across all examples.

meltsufin commented:
I think I've addressed all of the comments. @joaoandremartins If you still think something is off, feel free to update the branch and merge it without me, while I'm away. Thanks!
